### DISCUSSION TITLE
Error when two requests are too close in time

### DISCUSSION THREAD

**Discussion by diego09310** [spotify-web-api-java/spotify-web-api-java#367] [Q&A] [answered]
Hi, I'm getting this error when I make a request close after another (not always, it's a range, not inmediate and not long after it):
```
java.lang.IllegalStateException: Connection 192.168.1.152:43204<->35.186.224.25:443 is still allocated
    at org.apache.hc.core5.util.Asserts.check(Asserts.java:50) ~[httpcore5-5.2.2.jar:5.2.2]
    at org.apache.hc.client5.http.impl.io.BasicHttpClientConnectionManager.getConnection(BasicHttpClientConnectionManager.java:306) ~[httpclient5-5.2.1.jar:5.2.1]
...
```
From what I read this happens when using the BasicHttpClientConnectionManager and having more than one request open, depending on timing it will reuse the connection on create a new one.
I'm making the calls using the synchronous execute, but using spring boot so executions are probably in different threads (I also tried with async calls and got the same issue).

Is there a way to avoid this (I may be using the API wrong)? Right now I have a wrokaround with a wrapper that catches the exception and retries, but asking in case there's a better way to handle this.

Thanks

**Comment by dargmuesli** [comment, marked as answer]
You're on the right track. You can try to [use a pooling connection manager](https://stackoverflow.com/a/14866394/4682621) as `httpManager` by [building a custom SpotifyApi instance](https://spotify-web-api-java.github.io/spotify-web-api-java/se/michaelthelin/spotify/SpotifyApi.Builder.html).

**Reply by diego09310** [reply]
I thought about it, for now I'll keep it with the wrapper for simplicity but in the future I may try to implement it that way. Thanks!

**Reply by dargmuesli** [reply]
Ok! Also see this current PR: https://github.com/spotify-web-api-java/spotify-web-api-java/pull/375
